Technico/Services/PropertyService.cs

```diff
@@ -0,0 +1,112 @@
+﻿using Technico.Models;
+using Technico.Repositories;
+using Technico.Dtos;
+
+namespace Technico.Services;
+
+public class PropertyService 
+{
+    private readonly PropertyRepository _propertyRepository;
+
+    public PropertyService(PropertyRepository propertyRepository)
+    {
+        _propertyRepository = propertyRepository;
+    }
+
+    public async Task<PropertyDTO?> CreateAsync(PropertyDTO propertyDTO)
+    {
+        Console.WriteLine($"Input PropertyDTO - Address: {propertyDTO.Address}, OwnerID: {propertyDTO.OwnerID}");
+
+        var properties = await _propertyRepository.GetAllAsync();
+        bool PropertyExists = properties.Any(u => u.Address == propertyDTO.Address);
+        if (PropertyExists) return null;
+
+        var property = new Property
+        {
+            Address = propertyDTO.Address,
```

Comment:
Add checks to ensure required fields (e.g., Address, YearOfConstruction, OwnerID) are valid before proceeding with repository calls. Samegoes for the rest of the fields in every service